frontend/src/pages/chat/components/delete-chat-modal/delete-chat-modal.tsx

```diff
@@ -0,0 +1,67 @@
+import { forwardRef } from 'react';
+
+import { Button, Modal } from '#libs/components/components.js';
+import { DataStatus } from '#libs/enums/enums.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+} from '#libs/hooks/hooks.js';
+import { actions as chatActions } from '#slices/chats/chats.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  id: number | null;
+};
+
+const DeleteChatModal: React.ForwardRefRenderFunction<
+  HTMLDialogElement,
+  Properties
+> = ({ id }, reference) => {
+  const chatDeleteModalReference =
+    reference as React.RefObject<HTMLDialogElement | null>;
+  const dispatch = useAppDispatch();
+  const handleCancel = useCallback(() => {
+    chatDeleteModalReference.current?.close();
+  }, [chatDeleteModalReference]);
+
+  const { isLoading } = useAppSelector(({ chats }) => {
+    return {
+      isLoading: chats.deleteChatDataStatus === DataStatus.PENDING,
+    };
+  });
+
+  const handleDeleteChatEntry = useCallback(() => {
+    void dispatch(chatActions.deleteChat(id as number)).finally(() => {
+      chatDeleteModalReference.current?.close();
```

Comment:
It won't work, you have to close it on `FULFILLED` data status

Reply:
If I am not mistaken, to make it work we should add `.unwrap()` after `dispatch()`. But if there is a more clear solution of course we should try it first

Reply:
I'm not sure how `unwrap` will work in case of rejection, but basically it's also a good option

Reply:
But if we have rejected status, I think we also need close it.

Reply:
I think in case of getting rejected status client should see error toast, but modal still have to be opened

Reply:
Ok, I can do it